Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1012,14 +1067,17 @@ protected function prepareClassReflectionForUsage(string $className): string
      * @throws ClassSchemaConstraintViolationException
      * @throws Exception
      * @throws InvalidClassException
-     * @throws ReflectionException
+     * @throws InvalidPropertyTypeException
+     * @throws InvalidValueObjectException
+     * @throws \ReflectionException
      */
     protected function reflectEmergedClasses(): void
     {
-        $classNamesToReflect = [];
-        foreach ($this->availableClassNames as $classNamesInOnePackage) {
-            $classNamesToReflect = array_merge($classNamesToReflect, $classNamesInOnePackage);
+        $availableClassnames = [];
+        foreach ($this->availableClassNames as $classNamesInPackage) {
+            $availableClassnames[] = $classNamesInPackage;
         }
+        $classNamesToReflect = array_merge([], ...$availableClassnames);
```

Comment:
okay that part got me first confused ^^ clever though :D

it first looked like a `Object.assign({}, copyFrom)` in javascript but its way better, we unpack all arrays and then merge them together to a flat thing

we can make that a little more simple even:

```suggestion
        // flatten nested array structure to a list of classes
        $availableClassnames = array_merge(...array_values($this->availableClassNames));
```

We dont need the `[]` as since 7.4.0 this function can be called without any parameter, and it will return empty array.

And we still need the `array_values` to prevent: Uncaught ArgumentCountError: array_merge() does not accept unknown named parameters